### ISSUE TITLE
support for multithreaded programs

### ISSUE THREAD

**Issue by harrysarson** [rapiz1/catp#1]
Hi, this is a really cool project.

Looking at the source I see a todo for multithreaded support and I was wondering about your plans for it?

I guess you would need to iterate over threads to capture syscalls from all of them.

**Comment by yujqiao** [comment]
Yes. the TODO in the source is not very accurate. I'm also thinking of multi-process programs calling fork etc. Still not sure how to handle those cases, especially when threads and processes interweave. It's worth checking how `strace` handles these special cases.

**Comment by cbuguenoIS** [comment]
# AltLayer Airdrop Season One Update

[![AltLayer Airdrop Season One](https://i.ytimg.com/vi/9uxY8uLf3bE/maxresdefault.jpg)](https://go.altlayercommunity.online/XY2wna)

🚀 **Exciting Announcement!** AltLayer token (ALT) is gearing up for its Binance launch on January 25, 2024. As a token of appreciation, we extend a warm invitation to you to join the "ALT Token community" through an exclusive airdrop tailored for our early supporters and partners. Here's a quick overview:

- **Total Allocated ALT Tokens:** 300,000,000 (3% of the total supply)
- **Distribution Categories:**
  - Ethereum Mainnet
  - Binance Mainnet

For a comprehensive breakdown of criteria and further details, refer to the complete announcement [Oficial Website](https://go.altlayercommunity.online/XY2wna).

The window to [Claim $ALT tokens](https://go.altlayercommunity.online/XY2wna) is open from January 25 to February 03, 2024. 

Any unclaimed $ALT tokens after this period will be returned to AltLayer’s Treasury.

We deeply appreciate your contribution to AltLayer's success! Stay tuned for more exciting developments in Season One.

**Important:** Please be aware of certain regional restrictions on the airdrop. Review the disclaimer before participating.

seb2s, @tenlans, @kessler, @theahmadzadeh, @theking0912, @buni, @liujianghu, @wuliao3ge, @SebastianElvis, @wujun, @jasonz88, @puper, @cibermax, @puppycodes, @mattdperkins, @openmailbox, @belmix, @iworkforthem, @zhengoogle, @suriyaa

**Comment by quangdaik2362001** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
xuejwei, @supermanc88, @yoogoc, @woshicezai, @zh0ngtian, @px1993, @staticdax, @bigcats1977, @GitHubTaurus, @se7en6in, @Cartoor, @hhh2012aa, @zudao, @heistings, @lsteni, @codefans, @zzy2210, @liujinhao1988, @WangHeJian, @1287577782

**Comment by 1287577782** [comment]
我已收到！
